### PR TITLE
Made DateFormat class Threadsafe using ThreadLocale - Trunk- 4614

### PR DESCRIPTION
**Pull request by sashrika**
https://issues.openmrs.org/browse/TRUNK-4614

### CODE REVIEW

**Comment by rkorytkowski** [comment]
Could you please update the code to the latest version of master?

**Comment by dkayiwa** [comment]
@sashrika do you mind pulling from upstream and try resolve these conflicts?

File: api/src/main/java/org/openmrs/Obs.java [hunk 1]
```diff
@@ -1080,11 +1080,31 @@ public String getValueAsString(Locale locale) {
 		return "";
 	}
 	
-	private static DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+	private static ThreadLocal<DateFormat> dateFormat = new ThreadLocal<DateFormat>() { // Made DateFormat Threadsafe using ThreadLocal
```
Comment (djazayeri) [inline api/src/main/java/org/openmrs/Obs.java]: Is this better than just replacing the static dateFormat, timeFormat, and datetimeFormat fields with calls to: new SimpleDateFormat(...)? E.g. is introducing more performant than just instantiating these as-needed? (e.g. OpenmrsUtil.getDateFormat methods instantiate new ones...)

Comment (rkorytkowski) [inline api/src/main/java/org/openmrs/Obs.java]: Many google results e.g. http://stackoverflow.com/a/6840897 say ThreadLocal is the fastest option, but I haven't seen exact numbers quoted.

Comment (rkorytkowski) [inline api/src/main/java/org/openmrs/Obs.java]: Found tests at http://www.javacodegeeks.com/2010/07/java-best-practices-dateformat-in.html

**Comment by dkayiwa** [comment]
@sashrika remind me to reopen if you resume working on the above.

**Pull request closed without merging**